src/transactionprivate.cpp

```diff
@@ -49,29 +50,35 @@ void TransactionPrivate::setup(const QDBusObjectPath &transactionId)
                                                          tid.path(),
                                                          QDBusConnection::systemBus(),
                                                          q);
+
     QStringList hints = Daemon::global()->hints();
     hints << QStringLiteral("supports-plural-signals=true");
     q->setHints(hints);
 
     q->connect(p, SIGNAL(Destroy()), SLOT(destroy()));
 
     // Get current properties
-    QDBusMessage message = QDBusMessage::createMethodCall(PK_NAME,
-                                                          tid.path(),
-                                                          DBUS_PROPERTIES,
-                                                          QLatin1String("GetAll"));
-    message << PK_TRANSACTION_INTERFACE;
-    QDBusConnection::systemBus().callWithCallback(message,
-                                                  q,
-                                                  SLOT(updateProperties(QVariantMap)));
-
-    // Watch for properties updates
-    QDBusConnection::systemBus().connect(PK_NAME,
-                                         tid.path(),
-                                         DBUS_PROPERTIES,
-                                         QLatin1String("PropertiesChanged"),
+    auto props = new OrgFreedesktopDBusPropertiesInterface(PK_NAME, tid.path(), QDBusConnection::systemBus(), q);
+    Q_ASSERT(props->isValid());
+    QDBusPendingReply<QVariantMap> allPropertiesReply = props->GetAll(PK_TRANSACTION_INTERFACE);
+    QDBusPendingCallWatcher *w = new QDBusPendingCallWatcher(allPropertiesReply);
+    QObject::connect(w, &QDBusPendingCallWatcher::finished, q, [this, allPropertiesReply] (QDBusPendingCallWatcher *w) {
+        w->deleteLater();
+        if (w->isError()) {
+            qCWarning(PACKAGEKITQT_TRANSACTION) << "Failed to get properties from" << PK_NAME << w->error();
+        } else {
+            updateProperties(allPropertiesReply.value());
+        }
+    });
+
+    // Uncomment this next line to make it blocking
+    // allPropertiesReply.waitForFinished();
```

Comment:
Is this ever used? Would it make sense to just remove that comment, or hide the blocking mode behind an extra flag, in case API users actually want to have this blocking-wait for a reply?

Reply:
Yes, it should be removed, I only added it to help https://github.com/PackageKit/PackageKit-Qt/issues/42#issuecomment-1376575657 figure out their issue.

Reply:
Okay - otherwise this PR looks fine (but I don't see how it helps with #42, but that discussion over there is quite long, so there's a huge chance that I'm just missing the point ^^ - I also don't see these crashes, but there are multiple users affected on Debian, based on the bug reports we get).